Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -3337,6 +3430,7 @@ def _http_request(self, method, url_suffix, full_url=None, headers=None,
                 address = full_url if full_url else urljoin(self._base_url, url_suffix)
                 headers = headers if headers else self._headers
                 auth = auth if auth else self._auth
+                self._implement_retry(retries, status_list_to_retry, backoff_factor, raise_on_redirect, raise_on_status)
```

Comment:
lets add a condition here, and only if `retries` parameter is provided - greater than 1 then the code will go into  `_implement_retry`.
This way all the existing code will not use that code (which potentially can introduce new bugs).

Reply:
That's actually is not a good idea.

The implementation changes the session's configurations and the session is the class's attribute, not the method.

So assuming I did one call with retry configuration- the next call I'll do with retries=0 will also retry like the previous call because the configuration part that should reset the retries is not performed now.

In any case- setting retries to 0 will fail the call after the first failure and will cause no background changes.